### PR TITLE
Add `aromatic_matching=False` to isometry check

### PR DESCRIPTION
**Pull request by mattwthompson**
## Description

Taking @lilyminium's suggestion in #255 and just seeing what happens in tests.

Not squatting on this at all; if somebody wants to move this forward please feel free to use this branch/PR.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] Add `aromatic_matching=False` to isometry check
  - [x] Add test reproducing #255

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by mattwthompson** [comment]
Just a note to myself: adding a test like `test_componentresult_deduplication_iso` (or some others in the same file) with an appropriate molecule should cover this change. IIUC the key behavior is that two molecules (or two representations of, or two tautomers of, depending on your view)

* have the same InChi key
* are viewed by the toolkit as isomorphic when `aromatic_matching=False`
* are not viewed by the toolkit as isomorphic when `aromatic_matching=True` (the current behavior, and the toolkit's default)

The molecule in the linked issue appears to trigger these conditions via having a proton hop around within an aromatic ring

File: openff/qcsubmit/_tests/test_datasets.py [hunk 1]
```diff
@@ -343,6 +343,38 @@ def test_componentresult_deduplication_iso():
...
+    mol2.generate_conformers(n_conformers=1)
+
+    # this function returns a boolean equal to "is this molecule already in this dataset"
+    assert not result.add_molecule(mol1)
+    assert not result.add_molecule(mol2)
+
+    # ... and so the expected behavior is that they're deduplicated into the same
+    # "molecule" but the separate `Molecule` objects exist as distinct conformers
+    assert result.n_molecules == 1
+    assert result.n_conformers == 2
```
Comment (mattwthompson) [inline openff/qcsubmit/_tests/test_datasets.py:375]: I do not understand how this class and method are meant to behave, so everything I've written here should be fact-checked. This test is mostly cribbed from `test_componentresult_deduplication_iso ` above and the shifting boundaries of molecule identity make it hard to understand (at least for me at this particular moment). Brownie points if `ComponentResult.add_molecule` can be updated in a way that I understand what's going on

**Comment by mattwthompson** [comment]
Test added! If I revert the `aromatic_matching=False` change this test fails; with the change, the test passes.

File: openff/qcsubmit/_tests/test_datasets.py [hunk 2]
```diff
@@ -343,6 +343,38 @@ def test_componentresult_deduplication_iso():
     assert result.n_conformers == 2
 
 
+def test_componentresult_deduplication_tautomer():
+    """Reproduce issue #260 using molecules in the reported issue."""
```
Comment (j-wags) [inline openff/qcsubmit/_tests/test_datasets.py:347]:
```suggestion
    """Reproduce issue #255 using molecules in the reported issue."""
```

File: openff/qcsubmit/_tests/test_datasets.py (hunk 1 above)
Comment (j-wags) [reply openff/qcsubmit/_tests/test_datasets.py:375]: I agree with you assessment that the second call to `add_molecule` should return `True`. Working on updating that now.

**Review by j-wags — APPROVED** [review]
Thanks for making the great test. I also had your confusion about the return value of `ComponentResult.add_molecule`, and I realized it was clearly a bug, since instead of returning a `bool`, it would return `None`, and `assert(None)` quacked like `assert(False)`. I've fixed that and now everything seems hunky dorey, and I've added a little more testing for it as well. I've updated the release notes saying as much.

PS in a previous commend you said 

> The molecule in the linked issue appears to trigger these conditions via having a proton hop around within an aromatic ring

and I'll clarify for future readers that it's a double bond/formal charge that's hopping around, not a proton!

**Comment by jthorton** [comment]
Great job guys!

**Pull request merged**